src/main/java/com/likelion/backendplus4/talkpick/batch/news/article/infrastructure/collector/support/mapper/implement/DongaRssMapper.java

```diff
@@ -49,93 +59,109 @@ protected ScraperFactory getScraperFactory() {
+        ContentScraper scraper = getScraperFactory().getScraper(getMapperType())
+                .orElseThrow(() -> new ArticleCollectorException(ArticleCollectorErrorCode.SCRAPER_NOT_FOUND));
+
+        String scrapedContent = scraper.scrapeContent(link);
+        if (scrapedContent == null || scrapedContent.isEmpty()) {
+            throw new ArticleCollectorException(ArticleCollectorErrorCode.EMPTY_ARTICLE_CONTENT);
+        }
+
+        return removeUnwantedPhrases(scrapedContent);
     }
 
     /**
-     * RSS description에서 HTML 태그를 제거하고 문단을 추출하여 직렬화
+     * 동아일보 링크에서 고유 ID 추출
      *
-     * @param entry RSS 항목
-     * @return 직렬화된 문단 JSON 또는 원본 description
+     * @param link 기사 링크
+     * @return 추출된 고유 ID
+     * @throws ArticleCollectorException 링크가 null이거나 ID를 추출할 수 없는 경우
+     * @since 2025-05-10
      */
     @Override
-    protected String extractDescription(SyndEntry entry) {
-        if (entry.getDescription() == null) {
-            return "";
+    protected String extractUniqueIdFromLink(String link) {
+        if (link == null || link.trim().isEmpty()) {
+            throw new ArticleCollectorException(ArticleCollectorErrorCode.ARTICLE_ID_EXTRACTION_ERROR);
         }
 
-        String rawDescription = entry.getDescription().getValue();
-        if (rawDescription == null || rawDescription.isEmpty()) {
-            return "";
+        String[] parts = link.split("/");
+        if (parts.length >= 2) {
+            String id = parts[parts.length - 2];
+            if (id != null && !id.trim().isEmpty()) {
+                return id;
+            }
         }
 
-        return processHtmlContent(rawDescription);
+        throw new ArticleCollectorException(ArticleCollectorErrorCode.ARTICLE_ID_EXTRACTION_ERROR);
     }
 
     /**
-     * HTML 콘텐츠 처리하여 정제된 문단 직렬화
+     * 불용어 제거 메서드
+     * 저작권 문구, 광고 문구 등 불필요한 문구 제거
      *
-     * @param htmlContent HTML 콘텐츠
-     * @return 직렬화된 문단 JSON 또는 태그가 제거된 텍스트
+     * @param content 원본 내용
+     * @return 불용어가 제거된 내용
+     * @since 2025-05-17
      */
-    private String processHtmlContent(String htmlContent) {
-        try {
-            List<String> paragraphs = extractCleanParagraphs(htmlContent);
-            return serializeParagraphs(paragraphs);
-        } catch (Exception e) {
-            return removeAllHtmlTags(htmlContent);
+    private String removeUnwantedPhrases(String content) {
+        if (content == null || content.isEmpty()) {
+            return "";
         }
+
+        content = content.replaceAll("\\(c\\)\\s*동아일보", "");
+        content = content.replaceAll("저작권자.*동아일보.*무단.*전재.*금지", "");
+        content = content.replaceAll("무단전재 및 재배포 금지", "");
+        content = content.replaceAll("\\S+기자\\s+\\S+@donga\\.com", "");
+        content = content.replaceAll("동아닷컴 뉴스스탠드", "");
+        content = content.replaceAll("동아일보 홈페이지", "");
+        content = content.replaceAll("PARAGRAPH_BREAKPARAGRAPH_BREAK", "PARAGRAPH_BREAK");
+
+        return content.trim();
```

Comment:
P2:
개인적으로는 이러한 패턴들을 위로 올려 한곳에 모아놓는 것이 좋다고 생각합니다.
아래와 같이 람다식으로의 변경을 고려해보세요!

```suggestion
// 미리 모든 패턴을 정의해놓음
private static final List<String> UNWANTED_PATTERNS = List.of(
    "\\(c\\)\\s*동아일보",
    "저작권자.*동아일보.*무단.*전재.*금지",
    "무단전재 및 재배포 금지",
    "\\S+기자\\s+\\S+@donga\\.com",
    "동아닷컴 뉴스스탠드",
    "동아일보 홈페이지",
    "PARAGRAPH_BREAKPARAGRAPH_BREAK"
);

protected String removeUnwantedPhrases(String content) {
        // null 체크
    return Optional.ofNullable(content)
        // 빈 문자열 체크
        .filter(c -> !c.isEmpty())
        // .reduce(누적값을 취급할 때 사용하는 람다)
        // c(content 값을  UNWANTED_PATTERNS값을 이용하여 누적값을 취급하겠다.
        // c에 대하여 (누적값, 정규표현식) -> 누적값에서 정규표현식이 일치하면 모두 ""로 바꿔주겠다.
        .map(c -> UNWANTED_PATTERNS.stream()
            .reduce(c, (current, regex) -> current.replaceAll(regex, "")))
        // 앞뒤 공백 제거
        .map(String::trim)
        .orElse("");
}

```

Reply:
넵 위에 알려주신 코드랑 비슷한맥락인거같고 수정했습니다